docs/source/run_model_in_simulation.rst

```diff
@@ -190,15 +230,35 @@ block, results are printed::
           Aladdin Results
     ===============================
 
-Depending on the machine on which the simulation runs, it takes about 10 mins
-on my i7-9850H CPU to finish the simulation.
+How long the simulation will take depends on the machine. On my Intel Core i7
+9850H, it takes about 10 minutes. For larger models, both the trace storage and
+simulation time can become problematic, read on to see the solution.
 
-And congratulations! You just finished running the first SMAUG simulation. The
-:code:`outputs` folder contains simulation stats generated by gem5-Aladdin.
+.. _label_apply_sampling:
+
+Apply sampling to reduce simulation time and trace storage
+----------------------------------------------------------
+
+To save simulation time and trace storage for large models, we can sample the
+loop iterations in the oftentimes repetitive DL kernels. To enable sampling,
+use the :code:`--sample-level` and :code:`--sample-num` parameters in SMAUG.
+:code:`--sample-level` sets the sampling level, which can be one of the five
+levels: :code:`no`, :code:`low`, :code:`medium`, :code:`high` and
+:code:`very_high`. :code:`--sample-num` sets the number of sample iterations
+used by every sampling enabled entity, fewer sample iterations means more
+sampling. Details about using the sampling parameters and the gem5-Aladdin
+sampling APIs are discussed in `Sampling of accelerated kernels
+<doxygen_html/simulation.html>`_.
 
-Apply sampling to reduce simulation time
-----------------------------------------
+.. _label_multithreading:
 
-For large models, the trace storage and simulation time can become problematic.
-To solve these issue, we use sampling techniques detailed in `C++ side
-tutorials <doxygen_html/index.html>`_.
+Simulate with multiple threads
+------------------------------
+
+If the SoC has multiple CPUs (specified via the :code:`--num-cpus` parameter),
+we can enable SMAUG's multithreading feature by passing the
+:code:`--num-threads` parameter to it. See `Multithreading in gem5 SE mode
+<doxygen_html/simulation.html>`_ for more details.
+
+Congratulations! You just finished running the first SMAUG simulation. The
```

Comment:
Let's put the congrats earlier on, before discussion of sampling/multithreading.

Reply:
Done.